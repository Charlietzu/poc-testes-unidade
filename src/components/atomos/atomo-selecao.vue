<template>
  <v-select
    :items="itens"
    v-model="itemInterno"
    :item-text="textoItem"
    :item-value="valorItem"
    :placeholder="placeholder"
    dense
    outlined
    item-color="primary"
    @change="onChange"
    no-data-text="Os dados não estão disponíveis"
  ></v-select>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { Prop, Emit, Watch } from "vue-property-decorator";

@Component
export default class AtomoSelecao extends Vue {
  private itemInterno: number | null = null;

  @Watch("itemInterno")
  alterarItem(): void {
    this.itemInterno = this.itemSelecionado;
  }

  created(): void {
    this.itemInterno = this.itemSelecionado;
  }

  @Watch("itemInterno")
  @Emit("alteracaoSelect")
  onChange(valorSelecionado: number): number {
    return valorSelecionado;
  }

  @Prop()
  itemSelecionado!: number | null;

  @Prop()
  itens!: any[];

  @Prop()
  textoItem!: string;

  @Prop()
  valorItem!: number;

  @Prop({ type: String, default: "" })
  placeholder?: string;
}
</script>

<style lang="scss" scoped></style>
