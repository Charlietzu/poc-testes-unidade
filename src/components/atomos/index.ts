import AtomoSelecao from "./atomo-selecao.vue";

export { AtomoSelecao };
