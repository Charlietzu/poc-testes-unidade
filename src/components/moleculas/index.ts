import MoleculaSelecao from './molecula-selecao.vue';

export { MoleculaSelecao };
