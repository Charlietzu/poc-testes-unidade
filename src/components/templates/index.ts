import TemplatePadrao from "./template-padrao.vue";

export { TemplatePadrao };
