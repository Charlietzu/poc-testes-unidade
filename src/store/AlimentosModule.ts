import { Action, Module, Mutation, VuexModule } from "vuex-module-decorators";

export interface Alimento {
  nomeAlimento: string;
  codigoAlimento: number;
}

@Module({ name: "Alimentos" })
export default class AlimentosModule extends VuexModule {
  private _alimentos: Alimento[] = [
/*     {
      nomeAlimento: "Alface",
      codigoAlimento: 0,
    },
    {
      nomeAlimento: "Arroz",
      codigoAlimento: 1,
    },
    {
      nomeAlimento: "Batata",
      codigoAlimento: 2,
    }, */
  ];

  private _alimentoSelecionado: number | null = null;

  get alimentos(): Alimento[] {
    return this._alimentos;
  }

  get alimentoSelecionado(): number | null {
    return this._alimentoSelecionado;
  }

  @Mutation
  private setAlimentos(alimentos: Alimento[]): void {
    this._alimentos = alimentos;
  }

  @Mutation
  private setAlimentoSelecionado(codigoAlimento: number): void {
    this._alimentoSelecionado = codigoAlimento;
  }

  @Action
  atualizarAlimentoSelecionado(codigoAlimento: number): void {
    this.context.commit("setAlimentoSelecionado", codigoAlimento);
  }

  @Action
  obterAlimentos(alimentos: Alimento[]): void {
    this.context.commit("setAlimentos", alimentos);
  }
}
