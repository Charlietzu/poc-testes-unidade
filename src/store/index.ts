import Vue from "vue";
import Vuex from "vuex";
import AlimentosModule from "./AlimentosModule";
import { getModule } from "vuex-module-decorators";

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Alimentos: AlimentosModule,
  },
});

export default store;

const moduloAlimentos = getModule(AlimentosModule, store);

export { moduloAlimentos };
