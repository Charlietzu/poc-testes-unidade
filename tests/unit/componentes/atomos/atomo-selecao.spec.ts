import { createLocalVue, mount } from "@vue/test-utils";
import { AtomoSelecao } from "@/components/atomos";
import { Alimento } from "@/store/AlimentosModule";
import Vuetify from "vuetify";
import Vue, { VueConstructor } from "vue";

const localVue: VueConstructor<Vue> = createLocalVue();
describe("atomo-selecao.vue", () => {
  let vuetify: Vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const montarComponente = (options: Record<string, unknown>) => {
    return mount(AtomoSelecao, {
      localVue,
      vuetify,
      ...options,
    });
  };

  const alimentos: Alimento[] = [
    {
      nomeAlimento: "Alface",
      codigoAlimento: 0,
    },
    {
      nomeAlimento: "Arroz",
      codigoAlimento: 1,
    },
    {
      nomeAlimento: "Batata",
      codigoAlimento: 2,
    },
  ];

  const propsCustomizadas = {
    itens: alimentos,
    itemSelecionado: alimentos[0].codigoAlimento,
    valorItem: "codigoAlimento",
    textoItem: "nomeAlimento",
    placeholder: "Selecione Alimento",
  };
  it("Renderiza o componente corretamente", () => {
    //Arrange & Act
    const wrapper = montarComponente({
      propsData: propsCustomizadas,
    });

    //Assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renderiza o estado inicial corretamente", () => {
    //Arrange & Act
    const wrapper: any = montarComponente({
      propsData: propsCustomizadas,
    });

    //Assert
    expect(wrapper.vm.itemInterno).toBe(wrapper.vm.itemSelecionado);
  });

  it("Evento alteracaoSelect deve ser emitido com apenas um argumento", () => {
    //Arrange
    const wrapper: any = montarComponente({
      propsData: propsCustomizadas,
    });

    //Act
    wrapper.vm.onChange(1);

    //Assert
    expect(wrapper.emitted().alteracaoSelect.length).toBe(1);
  });

  it("Item interno deve ser igualado a Item selecionado quando sofrer alguma alteração", async () => {
    //Arrange
    const wrapper: any = montarComponente({
      propsData: propsCustomizadas,
    });

    //Act
    wrapper.setData({ itemInterno: Math.random() });
    await wrapper.vm.$nextTick();

    //Assert
    expect(wrapper.vm.itemInterno).toEqual(wrapper.vm.itemSelecionado);
  });
});
