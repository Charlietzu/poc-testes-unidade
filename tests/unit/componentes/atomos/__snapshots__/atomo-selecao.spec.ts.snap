// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atomo-selecao.vue Renderiza o componente corretamente 1`] = `
<div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder v-select">
  <div class="v-input__control">
    <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-2" class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">â€‹</span></legend>
      </fieldset>
      <div class="v-select__slot">
        <div class="v-select__selections">
          <div class="v-select__selection v-select__selection--comma">Alface</div><input id="input-2" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
        </div>
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden" value="0">
      </div>
      <div class="v-menu">
        <!---->
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
